server/src/visitors/YmlFunctionVisitor.ts

```diff
@@ -133,4 +148,92 @@ export class YmlFunctionVisitor extends YmlBaseVisitor {
+    }
+
+    public visitInstruction_for(node: Instruction_forContext) {
+        // `for` loop: complexity +1 + nesting level; increase nesting level for inner instructions
+        this.goDeeperInComplexity(node);
+    }
+
+    public visitInstruction_forall(node: Instruction_forallContext) {
+        // `forall` loop: complexity +1 + nesting level; increase nesting level for inner instructions
+        this.goDeeperInComplexity(node);
+    }
+
+    public visitInstruction_switchCase_asIf(node: Instruction_switchCase_asIfContext) {
+        // `switch {}`: complexity +1 + nesting level; increase nesting level for inner instructions
+        this.goDeeperInComplexity(node);
+    }
+    public visitInstruction_switchCase_withValue(node: Instruction_switchCase_withValueContext) {
+        // `switch(value) {}`: complexity +1 + nesting level; increase nesting level for inner instructions
+        this.goDeeperInComplexity(node);
+    }
+
+    public visitInstruction_switchExpr_asIf(node: Instruction_switchExpr_asIfContext) {
+        // `switchExpr {}`: complexity +1 + nesting level; increase nesting level for inner instructions
+        this.goDeeperInComplexity(node);
+    }
+    public visitInstruction_switchExpr_withValue(node: Instruction_switchExpr_withValueContext) {
+        // `switchExpr(value) {}`: complexity +1 + nesting level; increase nesting level for inner instructions
+        this.goDeeperInComplexity(node);
+    }
+
+    public visitIfExprBlock(node: IfExprBlockContext) {
+        // `ifExpr`: complexity +1 + nesting level; increase nesting level for inner instructions
+        this.goDeeperInComplexity(node);
+    }
+
+    public visitInstruction_break(_node: Instruction_breakContext) {
+        // `break` instruction: complexity ++
+        this.func.increaseCognitiveComplexity(false);
+    }
+
+    public visitCombinedCondition(node: CombinedConditionContext): void {
+        // sequence of binary logical operators: complexity ++
+        if (!!node.COND_AND() || !!node.COND_OR()) {
+            if (this.previousOperatorIsAnd === undefined || !(node._parent instanceof CombinedConditionContext)) {
+                this.previousOperatorIsAnd = !!node.COND_AND();
+            }
+            if (!!node.COND_AND() && !this.previousOperatorIsAnd) {
+                this.func.increaseCognitiveComplexity(false);
+            }
+            if (!!node.COND_OR() && !!this.previousOperatorIsAnd) {
+                this.func.increaseCognitiveComplexity(false);
+            }
+        }
+        super.visitCombinedCondition(node);
+    }
+
+    /**
+     * Increase the cognitive complexity of current AbstractYmlFunction
+     * then increase the nesting level and call the provided function or
+     * visit current node's children then decrease the nesting level.
+     *
+     * @param node currently visited node
+     * @param myFunction a function to apply after increasing the nesting level and before decreasing it
+     */
```

Comment:
~“Apply” belongs to the consumer vocabulary, i.e., functions with args. Here you just have a callable thingy with no arg.~
Even worse lol, it’s for functions with _in_ and _out_ stuff.
https://docs.oracle.com/javase/8/docs/api/java/util/function/Consumer.html#accept-T-
https://docs.oracle.com/javase/8/docs/api/java/util/function/Function.html#apply-T-

Reply:
What about `call` instead of `apply`?

Reply:
Yeah, that fits what you do.